### DISCUSSION TITLE
Syntax highlighting and formatting .liq files

### DISCUSSION THREAD

**Discussion by jooola** [savonet/liquidsoap#1780] [Q&A]
I was looking for a way to improve my experience when working with liquid soap files.

How can I enable syntax highlighting on .liq files ?
I tried Ocaml syntax but this doesn't feel like it's the same.

How to format code in .liq files ?
Again I tried ocamlformat but this didn't work as expected.
I didn't try dune, maybe this one does the trick.

So you guessed it, I don't know much about ocaml, and it's ecosystem. I would need some help to get this to work !

**Comment by add-n2x** [comment]
which IDE are you using?

I'm on VSCode and Syntax Highlighting is pretty good i think.

**Comment by smimram** [comment]
There is also good support for Emacs.

The Liquidsoap language is *not* OCaml, it is a language on its own!

**Comment by jooola** [comment]
I am using codium (vscode), or vim. But doesn't seem to work out of the box, what should I install ?

For vim this ? https://github.com/mcfiredrill/vim-liquidsoap

But regarding code formatting, how should I do ?

**Comment by smimram** [comment]
For VScode is seems to be this: https://github.com/vittee/vscode-liquidsoap

**Comment by jooola** [comment]
Ok thanks, with codium you use the open-vsx repository for extensions, which require dev to publish there extensions in a extra repository. This is why I didn't find this easily.

And what about formatting liquidsoap code ? Is there any interest for such tool ?

**Comment by dikkedimi** [comment]
would love an equivalent for atom!

**Comment by dtcooper** [comment]
I usually use Ruby for when the liquidsoap highlighting plugin gets wonky in VSCode. It works pretty well.
